src/NuGet.Core/NuGet.Frameworks/FrameworkNameProvider.cs

```diff
@@ -993,8 +968,12 @@ public bool TryGetSubSetFrameworks(string frameworkIdentifier, out IEnumerable<s
             return false;
         }
 
-        public int CompareFrameworks(NuGetFramework x, NuGetFramework y)
+        public int CompareFrameworks(NuGetFramework? x, NuGetFramework? y)
         {
+            if (x is null && y is null) return 0;
+            if (x is null) return -1;
+            if (y is null) return 1;
```

Comment:
This method, and `CompareUsingPrecedence` below, would have previously thrown `NullReferenceExceptions` if either of the inputs were null. However, in versions of the BCL where `IComparable` is annotated, `Compare` methods allow nulls. So, in the spirit of "least surprise", I thought we should also support nulls. But, it is a behaviour change.